src/sinks/nats.rs

```diff
@@ -223,11 +228,13 @@ impl StreamSink<Event> for NatsSink {
                     finalizers.update_status(EventStatus::Delivered);
 
                     events_sent.emit(CountByteSize(1, event_byte_size));
-                    bytes_sent.emit(ByteSize(bytes.len()));
+                    bytes_sent.emit(ByteSize(message_size));
                 }
             }
         }
 
+        let _ = self.connection.flush().await;
+
```

Comment:
```suggestion
```

Reply:
The reason why I put it here is that you'll destroy your throughput if you wait for flushing to be done before writing the next message.

Reply:
That's a fair point. 🤔

Reply:
Ok. Take into account that Core NATS is an _at most once_ system, so flushing doesn't really give you many guarantees

Reply:
Hi. We would like to ensure the flush is successful before reporting a success. It ties in with Vector's end-to-end acknowledgements. Would it be possible to make the suggested changes? 

If performance does become an issue for users, we can revisit.

Reply:
Done